src/cpp/src/whisper/whisper_utils.hpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <openvino/openvino.hpp>
+
+#include "openvino/genai/perf_metrics.hpp"
+
+namespace ov {
+
+namespace genai {
+
+namespace utils {
+
+void infer_with_perf_metrics(ov::InferRequest& request, ov::genai::RawPerfMetrics& raw_metrics);
+
+template <typename T>
+void filter_by_ranges(std::vector<T>& value, size_t offset, std::vector<std::pair<size_t, size_t>>& ranges);
```

Comment:
It seems `filter_by_ranges` used only in `whisper_ustils.cpp`. If you don't want to use it somewhere else it might make sense do not expose it to `ov::genai::utils` namespace. You can use unnamed namespace in `whisper_utils.cpp` (example: https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/src/whisper/whisper.cpp#L23).